requirements.txt

```diff
@@ -9,3 +9,4 @@ python-dateutil>=2.8.0
 stevedore>=3.0.0
 typing-extensions; python_version<'3.11'
 symengine>=0.9,!=0.10.0
+pydantic
```

Comment:
Why is pydantic needed here? None of the RFCs outlined a use for this in any of the base primitive classes. I'm pretty wary of adding a new dependency just for input validation on options. I don't think we should be using it on any of the base interface definitions which the options rfc said too. If implementations want to use pydantic interfaces that should be compatible but it shouldn't be a hard requirement. Also using it just for the new statevector estimator implementation is kind of overkill to enforce that shots is an int and >= 0.  In general I'm a fan of pydantic as a library (especially compared to marshmallow which Qiskit's previously used it for a similar reason). But I really don't think it's something we should be adding it as a requirement here.

As an aside, this almost certainly has a lower bound since of at least >=2.0 since I'm pretty sure you're using pydantic v2 here. So you should probably have this set at `pydantic>=2`.

Reply:
[This RFC](https://github.com/Qiskit/RFCs/blob/master/0018-primitive-options-type.md) proposed using Python and **pydantic dataclasses** as the object type of primitive options. Of course, we could use Python's built-in dataclass, but since we are determined to use pydantic in a subclass (i.e. qiskit-ibm-runtime), I don't know if mixing the two is a good decision.
However, it is understandable to be cautious about adding new dependencies.
I'm very difficult to decide which one is better. Maybe it should be optional dependency to begin with.

Thanks for the pydantic version. It is indeed true.

Reply:
There are some other comments suggesting removing the use of dataclasses to help implement, eg, Pub. Since dataclasses aren't doing heavy lifting in those implementations (they give a repr), I am now in favour of making that change.

Reply:
Sorry if it wasn't clear in the RFC, which meant to say we'd use Python dataclasses for options in the base, and implementers can choose to use more advanced dataclasses like pydantic.